r/R/dataset.R

```diff
@@ -527,6 +527,9 @@ names.Dataset <- function(x) names(x$schema)
 #' @export
 dim.Dataset <- function(x) c(x$num_rows, x$num_cols)
 
+#' @export
+dimnames.Dataset <- function(x) list(NULL, names(x))
```

Comment:
We might consider a message here that tells folks that the `NULL` doesn't mean there are none. But then again dbplyr doesn't do that!

Reply:
Yeah, it's not immediately obvious, but Tables all inherit `ArrowTabular` which has `dimnames.ArrowTabular` defined already